runtime/major_gc.c

```diff
@@ -273,15 +273,114 @@ Caml_inline void mark_stack_push(struct mark_stack* stk, value block,
 }
 
 
+#ifdef NO_NAKED_POINTERS
+#ifdef NAKED_POINTERS_CHECKER
+
+#ifdef _WIN32
+#include <windows.h>
+
+static inline int safe_load(volatile value* p, value* result)
+{
+  value v;
+  __try {
+    v = *p;
+  }
+  __except(GetExceptionCode() == EXCEPTION_ACCESS_VIOLATION ?
+        EXCEPTION_EXECUTE_HANDLER : EXCEPTION_CONTINUE_SEARCH) {
+    *result = 0xdeadbeef;
+    return 0;
+  }
+  *result = v;
+  return 1;
+}
+#else
+static inline int safe_load (header_t * addr, /*out*/ header_t * contents)
+{
+  int ok;
+  header_t h;
+  intnat tmp;
+
+  asm volatile(
+      "leaq 1f(%%rip), %[tmp] \n\t"
+      "movq %[tmp], 0(%[handler]) \n\t"
+      "xorl %[ok], %[ok] \n\t"
+      "movq 0(%[addr]), %[h] \n\t"
+      "movl $1, %[ok] \n\t"
+  "1: \n\t"
+      "xorq %[tmp], %[tmp] \n\t"
+      "movq %[tmp], 0(%[handler])"
+      : [tmp] "=&r" (tmp), [ok] "=&r" (ok), [h] "=&r" (h)
+      : [addr] "r" (addr),
+        [handler] "r" (&(Caml_state->checking_pointer_pc)));
+  *contents = h;
+  return ok;
+}
+#endif
+
+#if SIZEOF_LONG == SIZEOF_PTR
+#define VAL_ONE 1ul
```

Comment:
I believe that the definition of `VAL_ONE` is not necessary, and can be replaced by `1ull` everywhere. @dra27

Reply:
I agree - I wrote it originally and only understood the problem after I committed 🙂

I think it should be OK to replace `VAL_ONE` with `INT64_LITERAL(1)`